bundles/org.eclipse.ui/src/org/eclipse/ui/internal/SearchDecoration.java

```diff
@@ -62,21 +62,55 @@ public static boolean validateRegex(String regex, ControlDecoration targetDecora
 	 * @return The appropriate error message if the regex is invalid or an empty
 	 *         string if the regex is valid.
 	 */
-	private static String getValidationError(String regex) {
+	private static String getValidationError(String regex, org.eclipse.swt.widgets.Control targetControl) {
+
 		try {
 			Pattern.compile(regex);
 			return ""; //$NON-NLS-1$
+
 		} catch (PatternSyntaxException e) {
-			String message = e.getLocalizedMessage();
+			return buildValidationErrorString(e, targetControl);
+		}
+	}
+
+	private static String buildValidationErrorString(PatternSyntaxException e, org.eclipse.swt.widgets.Control targetControl) {
+		GC gc = new GC(targetControl);
+
+		String description = e.getDescription();
+		int errorIndex = e.getIndex();
+		String pattern = e.getPattern();
+
+		if (errorIndex == -1) {
+			return description;
+		}
+		StringBuilder sBuilder = new StringBuilder();
+
+		sBuilder.append(description);
+		sBuilder.append(" at index ").append(errorIndex).append(System.lineSeparator()); //$NON-NLS-1$
+		sBuilder.append(pattern).append(System.lineSeparator());
+
+		String stringToIndexString = pattern.substring(0, errorIndex + 1);
+		String buildString = ""; //$NON-NLS-1$
```

Comment:
Get rid of this one, you don't need it. You need to `append` to the `StringBuilder` directly

Reply:
done